Modules/Module_Processing/Module_ASL/xASL_wrp_Quantify.m

```diff
@@ -205,7 +207,40 @@ function xASL_wrp_Quantify(x, PWI4D_Path, pathOutputCBF, M0Path, SliceGradientPa
 	x = rmfield(x, 'hematocrit');
 end
 
+% b. We model the expected hematocrit from age & sex
+% only if the Hematocrit was not specifically provided
+if ~isfield(x, 'Hematocrit')
+    IndexSetsAge = find(strcmpi(x.S.SetsName, 'age'));
+    IndexSetsSex = find(strcmpi(x.S.SetsName, 'sex'));
+    if ~isempty(IndexSetsAge) || ~isempty(IndexSetsSex)
+        fprintf('%s\n', 'Trying to infer hematocrit from age & sex');
+        
+        age = x.S.SetsID(:, IndexSetsAge);
+        
+        % Convert sex correctly
+        sex = x.S.SetsID(:, IndexSetsSex);
+        sexOptions = x.S.SetsOptions{:, IndexSetsSex};
+        sexN = zeros(length(sex), 1)
+        for iOption=1:length(sexOptions)
+            if ~isempty(regexpi(sexOptions{iOption}, '^(male|m|man|men)$'))
+                sexN(sex==iOption) = 1;
```

Comment:
I understand that this value is only used internally and that input in participants tsv is male/female. But it would be nicer to follow the statistical convention and define binary categorical variables like sex as 0/1 and not 1/2.

Reply:
Shall we restart the same discussion? ;)

You cannot use 0 as an index so this will not work with the current setup. The way it works now, it saves x.S.SetsOptions, and the value 1 points to x.S.SetsOptions{1} etc. x.S.SetsOptions{1} etc can be 0 1 or male female or 1 2 whatever you want. So for categorical variables (binary or three values or whatever) are all stored as indices in x.S.SetsID.